chalice/package.py

```diff
@@ -357,15 +386,255 @@ def _register_cfn_resource_name(self, name):
         return cfn_name
 
 
+class TerraformGenerator(TemplateGenerator):
+
+    template_file = "chalice.tf.json"
+
+    def generate(self, resources):
+        # type: (List[models.Model]) -> Dict[str, Any]
+        template = {
+            'resource': {},
+            'terraform': {
+                'required_version': '> 0.11.0'
+            },
+            'data': {
+                'aws_caller_identity': {'chalice': {}},
+                'aws_region': {'chalice': {}}
+            }
+        }
+        for resource in resources:
+            self.dispatch(resource, template)
+        return template
+
+    def _fref(self, lambda_function, attr='arn'):
+        # type: (models.ManagedModel, str) -> str
+        return '${aws_lambda_function.%s.%s}' % (
+            lambda_function.resource_name, attr)
+
+    def _arnref(self, arn_template, **kw):
+        # type: (str, str) -> str
+        d = dict(
+            region='${aws_region.chalice.name}',
```

Comment:
For these arn ref's, I think we need to use `data.aws_region` etc. As is terraform complains that `aws_region` and `aws_caller_identity` are not in the root module.